From 0a169f349bb3b8d73c3ee30c1f2207fabbbe2312 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 11:19:53 +0700
Subject: [PATCH 278/475] ARM: dts: r8a7743: Add VCP/VPC nodes

Add vcp/vpc nodes for uvcs-kernel-module with LTS-4.4 kernel support.

These are issues when porting from 3.10 to 4.4
	1. Cannot request interrupt for HW IP (VCP)
	2. Cannot map HW IP register (VPC)

These is a difference in register platform device for the driver:
	-Old method: driver use kernel API (platform_device_register())
	to register it device to kernel,
	-New method: devices are registered basing on the information
	provided by device tree.

Currently, only vcp0, vpc0 and prr are used.
vpc1, vpc0xy, and vpc1xy are added as a reserve.

The prr is used for getting LSI information which tell the driver
what kind of board is running.

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7743.dtsi | 48 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index 4efcd1e..7d16fa0 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -1752,6 +1752,54 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				<0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7743_CLK_VCP0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7743_CLK_VPC0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A7743_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A7743_CLK_VPC0>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A7743_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		vsp@fe928000 {
 			compatible = "renesas,vsp1";
 			reg = <0 0xfe928000 0 0x8000>;
-- 
2.7.4

